src/awesome/button.rs

```diff
@@ -0,0 +1,343 @@
+use std::fmt::{self, Display, Formatter};
+use std::default::Default;
+use std::rc::Rc;
+use rlua::{self, Table, Lua, UserData, ToLua, Value};
+use super::object::{Object, Objectable};
+use super::signal;
+use super::property::Property;
+use super::class::{self, Class};
+use rustwlc::types::KeyMod;
+use xcb::ffi::xproto::xcb_button_t;
+
+#[derive(Clone, Debug)]
+pub struct ButtonState {
+    button: xcb_button_t,
+    modifiers: KeyMod
+}
+
+#[derive(Clone, Debug)]
+pub struct Button<'lua>(Table<'lua>);
+
+impl <'lua> Button<'lua> {
+    pub fn button(&self) -> rlua::Result<Value<'lua>> {
+        let button = self.0.get::<_, ButtonState>("data")?;
+        Ok(Value::Integer(button.button as _))
+    }
+
+    pub fn set_button(&self, new_val: xcb_button_t) -> rlua::Result<()> {
+        let mut button = self.0.get::<_, ButtonState>("data")?;
+        button.button = new_val;
+        self.0.set("data", button)?;
```

Comment:
Double check with a test, but this should be unnecessary

Reply:
Nope, `UserData` is cloned out, no way to get around the forced copy. May revisit if it becomes annoying. Adding an update per `Drop` is stupid and not really possible. So I'll just accept it